pkg/codegen/schema.go

```diff
@@ -759,15 +760,26 @@ func generateUnion(outSchema *Schema, elements openapi3.SchemaRefs, discriminato
 		}
 
 		if discriminator != nil {
+			// Explicit mapping.
+			var mapped bool
 			for k, v := range discriminator.Mapping {
 				if v == element.Ref {
 					outSchema.Discriminator.Mapping[k] = elementSchema.GoType
+					mapped = true
 					break
 				}
 			}
+			// Implicit mapping.
+			if !mapped {
+				outSchema.Discriminator.Mapping[elementSchema.GoType] = elementSchema.GoType
```

Comment:
I don't think elementSchema.GoType should be used as a key here. It should be schema name (GoType is camel-cased and have a prefix)

Reply:
@Warboss-rus thank u!
It is my typo. Of course the schema name should be here. Added tests and fixed 👍